CMakeLists.txt

```diff
@@ -148,6 +148,7 @@ configure_package_config_file(cmake/mdspanConfig.cmake.in
 )
 write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/mdspanConfigVersion.cmake
   COMPATIBILITY SameMajorVersion
+  ARCH_INDEPENDENT
```

Comment:
https://github.com/kokkos/mdspan/blob/d861a3c6eb8cf5da2baa877d7eac114e7fd42453/CMakeLists.txt#L1
`ARCH_DEPENDENT` is a CMake 3.14 thing

Reply:
Right, sorry I missed that. Can I bump the minimum required version to 3.14 in this PR or would you rather keep supporting 3.13 in which case this PR can be declined?

Reply:
I have no objection to require a more recent version but I see that it might involve some update in the CI setup
https://github.com/kokkos/mdspan/commit/183cab273ceb356091dee60b5c3f936e4850b7e5

Reply:
Based on the comment above I went ahead and updated the minimum version to 3.14. It seems that all checks have passed.